src/mbgl/renderer/layers/render_fill_layer.cpp

```diff
@@ -521,21 +522,36 @@ void RenderFillLayer::update(gfx::ShaderRegistry& shaders,
                 return false;
             }
 
+            const auto& data = drawable.getData();
+            if (!data) return false;
+
             auto& uniforms = drawable.mutableUniformBuffers();
-            if (uniforms.get(FillLayerTweaker::idFillInterpolateUBOName)) {
-                uniforms.createOrUpdate(FillLayerTweaker::idFillInterpolateUBOName, &getFillInterpolateUBO(), context);
-            } else if (uniforms.get(idFillOutlineInterpolateUBOName)) {
-                uniforms.createOrUpdate(idFillOutlineInterpolateUBOName, &getFillOutlineInterpolateUBO(), context);
-            } else if (uniforms.get(idFillPatternInterpolateUBOName)) {
-                uniforms.createOrUpdate(idFillPatternInterpolateUBOName, &getFillPatternInterpolateUBO(), context);
-                uniforms.createOrUpdate(idFillPatternTilePropsUBOName, &getFillPatternTilePropsUBO(), context);
-            } else if (uniforms.get(idFillOutlinePatternInterpolateUBOName)) {
-                uniforms.createOrUpdate(
-                    idFillOutlinePatternInterpolateUBOName, &getFillOutlinePatternInterpolateUBO(), context);
-                uniforms.createOrUpdate(
-                    idFillOutlinePatternTilePropsUBOName, &getFillOutlinePatternTilePropsUBO(), context);
-            } else {
-                assert(false);
+            switch (static_cast<const gfx::FillDrawableData&>(*data).type) {
+                case gfx::FillVariant::Fill: {
+                    uniforms.createOrUpdate(
+                        FillLayerTweaker::idFillInterpolateUBOName, &getFillInterpolateUBO(), context);
+                    break;
+                }
+                case gfx::FillVariant::FillOutline: {
+                    uniforms.createOrUpdate(idFillOutlineInterpolateUBOName, &getFillOutlineInterpolateUBO(), context);
+                    break;
+                }
+                case gfx::FillVariant::FillPattern: {
+                    uniforms.createOrUpdate(idFillPatternInterpolateUBOName, &getFillPatternInterpolateUBO(), context);
+                    uniforms.createOrUpdate(idFillPatternTilePropsUBOName, &getFillPatternTilePropsUBO(), context);
+                    break;
+                }
+                case gfx::FillVariant::FillOutlinePattern: {
+                    uniforms.createOrUpdate(
+                        idFillOutlinePatternInterpolateUBOName, &getFillOutlinePatternInterpolateUBO(), context);
+                    uniforms.createOrUpdate(
+                        idFillOutlinePatternTilePropsUBOName, &getFillOutlinePatternTilePropsUBO(), context);
+                    break;
+                }
+                default: {
+                    assert(false);
+                    std::terminate();
```

Comment:
I don't think we do this anywhere else in the codebase. Not sure what the program behavior is if this is ever hit, but it can't be any worse than an apparant crash.

Reply:
Oops, addressed with a debug log.